File: be-service/app/models/demand_forecasting.py
from pydantic import BaseModel


class DemandForecastRequest(BaseModel):
    product_id: int
    horizon_days: int


class DemandForecastResponse(BaseModel):
    product_id: int
    forecast_quantity: int
    current_stock: int
    remaining_stock: int
    recommendation: str

File: fe-service/src/lib/demandForecasting.ts
import api from "./api";

export interface DemandForecastResponse {
  product_id: number;
  forecast_quantity: number;
  current_stock: number;
  remaining_stock: number;
  recommendation: string;
}

export async function generateDemandForecast(
  productId: number,
  horizonDays: number
): Promise<DemandForecastResponse> {
  const res = await api.post(
    "/api/v1/demand-forecasting/predict",
    {
      product_id: productId,
      horizon_days: horizonDays,
    }
  );

  return res.data.data;
}